Bottom: 1dae3f0c743763e0bac6a790ce8518578ccb28e9
Top:    e8bea7e5a050172cb24cb267d7872f0c38173f45
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-26 23:38:01 +0100

Add support for a two soldermask layers, defining regions of removed mask

File-format wise, these work similarly to how silk layers are supported now:

Layer (1 to n "[NAME]")     # "ordinary" design layers
Layer (n+1 "silk")          # top side
Layer (n+2 "silk")          # bottom side
Layer (n+3 "soldermask")    # top side
Layer (n+4 "soldermask")    # bottom side


---

diff --git a/src/const.h b/src/const.h
index 08fe253..2a5a683 100644
--- a/src/const.h
+++ b/src/const.h
@@ -48,6 +48,8 @@
  */
 #define BOTTOM_SILK_LAYER       0
 #define TOP_SILK_LAYER          1
+#define BOTTOM_SOLDERMASK_LAYER 2
+#define TOP_SOLDERMASK_LAYER    3
 
 /* ---------------------------------------------------------------------------
  * misc constants
diff --git a/src/data.h b/src/data.h
index 5d40d71..89d9b87 100644
--- a/src/data.h
+++ b/src/data.h
@@ -49,6 +49,8 @@ extern PCBType *PCB;
 #define max_copper_layer (PCB->Data->LayerN)
 #define bottom_silk_layer (max_copper_layer + BOTTOM_SILK_LAYER)
 #define top_silk_layer (max_copper_layer + TOP_SILK_LAYER)
+#define bottom_soldermask_layer (max_copper_layer + BOTTOM_SOLDERMASK_LAYER)
+#define top_soldermask_layer (max_copper_layer + TOP_SOLDERMASK_LAYER)
 
 extern SettingType Settings;
 
diff --git a/src/global.h b/src/global.h
index 1d017fe..bcbbe29 100644
--- a/src/global.h
+++ b/src/global.h
@@ -195,7 +195,7 @@ OutputType;
 typedef struct
 {
   Cardinal Number[MAX_LAYER],	/* number of entries per groups */
-    Entries[MAX_LAYER][MAX_LAYER + 2];
+    Entries[MAX_LAYER][MAX_LAYER + 4];
 } LayerGroupType;
 
 struct BoxType		/* a bounding box */
@@ -397,7 +397,7 @@ typedef struct			/* holds all objects */
   rtree_t *via_tree, *element_tree, *pin_tree, *pad_tree, *name_tree[3],	/* for element names */
    *rat_tree;
   struct PCBType *pcb;
-  LayerType Layer[MAX_LAYER + 2];	/* add 2 silkscreen layers */
+  LayerType Layer[MAX_LAYER + EXTRA_LAYERS];	/* add 2 silkscreen layers and 2 soldermask layers */
   int polyClip;
 } DataType;
